packages/react-components/src/components/grid.tsx

```diff
@@ -73,14 +74,17 @@ class Grid extends PureComponent<Props, State> {
     unmounted: boolean = false
     paginator = gifPaginator(this.props.fetchGifs, this.state.gifs)
     static getDerivedStateFromProps: GetDerivedStateFromProps<Props, State> = (
-        { columns, gutter, width }: Props,
+        { columns, gutter, width, externalGifs }: Props,
         prevState: State
     ) => {
         const gutterOffset = gutter * (columns - 1)
         const gifWidth = Math.floor((width - gutterOffset) / columns)
         if (prevState.gifWidth !== gifWidth) {
             return { gifWidth }
         }
+        if (externalGifs && externalGifs !== prevState.gifs) {
```

Comment:
if `externalGifs` exists, is this ever going to be `false`? i can't remember if `===` works with arrays of objects or not.